backend/src/Edna/Analysis/FourPL.hs

```diff
@@ -45,8 +60,49 @@ instance ToSchema Params4PL where
   declareNamedSchema Proxy =
     declareNamedSchema @Params4PLTuple Proxy
 
--- | This function performs actual analysis and finds the best parameters.
+type AnalysisResult = Either Text Params4PL
+--  deriving stock (Generic, Show, Eq)
+--
+--deriveJSON ednaAesonWebOptions ''AnalysisResult
+--
+--instance Buildable AnalysisResult where
+--  build = buildFromJSON
 --
--- TODO [EDNA-71] Implement!
-analyse4PL :: [(Double, Double)] -> IO Params4PL
-analyse4PL _ = pure (Params4PL 1 2 3 4)
+--instance ToSchema AnalysisResult where
+--  declareNamedSchema Proxy =
+--    declareNamedSchema @AnalysisResult Proxy
+
+data Params4PLReq = Params4PLReq
+  { plreqExperiment :: ExperimentId
+  , plreqData :: [(Double, Double)]
+  } deriving stock (Generic, Show, Eq)
+
+deriveJSON ednaAesonWebOptions ''Params4PLReq
+
+data Params4PLResp = Params4PLResp
+  { plrspExperiment :: ExperimentId
+  , plrspData :: AnalysisResult
+  } deriving stock (Generic, Show, Eq)
+
+deriveJSON ednaAesonWebOptions ''Params4PLResp
+
+-- | This function performs actual analysis and finds the best parameters.
+analyse4PL :: [(ExperimentId, [Measurement])] -> Edna [Params4PLResp]
```

Comment:
1. Why does this module need to know about `ExperimentId`s at all? Why can't it take a list of lists of measurements (maybe wrapped into a newtype) and return a list of `AnalysisResult`s?
2. Why `ExperimentId` rather than `SubExperimentId`?

Reply:
1. Hm, for real it doesn't need it. I just wanted to add some info to responses/requests and also to guarantee that it will be in the origin sequence (like when we have ids we can not think about it)
2. Because we don't know `ExperimentId` before computing it

So, I don't see any disadvantages to not use `ExperimentId` here

Reply:
> So, I don't see any disadvantages to not use `ExperimentId` here

1. You need to call `getExperimentId` in `analyseNewSubExperiment`. It's not expensive, but still redundant.
2. I still find it confusing that `ExperimentId`s are used here because we do analysis for sub-experiments, not for experiments.

That said, probably it can be put into "to be handled later" group, maybe I'll refactor it myself in a separate PR.